plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -2958,11 +2985,18 @@ private KVMPhysicalDisk getPhysicalDiskPrimaryStore(PrimaryDataStoreTO primaryDa
     protected void setDiskIoDriver(DiskDef disk) {
         if (getHypervisorLibvirtVersion() >= HYPERVISOR_LIBVIRT_VERSION_SUPPORTS_IO_URING
                 && getHypervisorQemuVersion() >= HYPERVISOR_QEMU_VERSION_SUPPORTS_IO_URING
-                && AgentPropertiesFileHandler.getPropertyValue(AgentProperties.ENABLE_IO_URING)) {
+                && isIoUringEnabled()) {
             disk.setIoDriver(DiskDef.IoDriver.IOURING);
         }
     }
 
+    /**
+     * IO_URING supported if it is supported by qemu AND the property 'enable.io.uring' is set to true
+     */
+    private boolean isIoUringEnabled() {
+        return isIoUringSupportedByQemu() && AgentPropertiesFileHandler.getPropertyValue(AgentProperties.ENABLE_IO_URING);
```

Comment:
Should this be `propertyValue || if-io-uring-is-supported()` ?

Reply:
IMHO if "ldd qemu" does not reveal uring associations, but "enable.io.uring=true" is enabled in the agent.properties, then Cloudstack should still attempt to use uring. This is to cover custom (maybe statically linked etc) qemu-kvm installations.